collectors/python.d.plugin/uwsgi/uwsgi.chart.py

```diff
@@ -0,0 +1,169 @@
+
+# NOTE: lines are created dynamically in `check()` method
+CHARTS = {
+    'requests': {
+        'options': [None, 'Requests', 'requests/s', 'requests', 'uwsgi.requests', 'stacked'],
+        'lines': [
+            ['requests', 'requests', 'incremental']
+        ]
+    },
+    'tx': {
+        'options': [None, 'Transmitted data', 'KB/s', 'requests', 'uwsgi.tx', 'stacked'],
+        'lines': [
+            ['tx', 'tx', 'incremental']
+        ]
+    },
+    'avg_rt': {
+        'options': [None, 'Average request time', 'ms', 'requests', 'uwsgi.avg_rt', 'line'],
+        'lines': [
+            ['avg_rt', 'avg_rt', 'absolute']
+        ]
+    },
+    'memory_rss': {
+        'options': [None, 'RSS (Resident Set Size)', 'MB', 'memory', 'uwsgi.memory_rss', 'stacked'],
+        'lines': [
+            ['memory_rss', 'memory_rss', 'absolute', 1, 1024 * 1024]
+        ]
+    },
+    'memory_vsz': {
+        'options': [None, 'VSZ (Virtual Memory Size)', 'MB', 'memory', 'uwsgi.memory_vsz', 'stacked'],
+        'lines': [
+            ['memory_vsz', 'memory_vsz', 'absolute', 1, 1024 * 1024]
+        ]
+    },
+    'exceptions': {
+        'options': [None, 'Exceptions', 'exceptions', 'exceptions', 'uwsgi.exceptions', 'line'],
+        'lines': [
+            ['exceptions', 'exceptions', 'incremental']
+        ]
+    },
+    'harakiri': {
+        'options': [None, 'Harakiris', 'harakiris', 'harakiris', 'uwsgi.harakiris', 'line'],
+        'lines': [
+            ['harakiri_count', 'harakiris', 'incremental']
+        ]
+    },
+    'respawn': {
+        'options': [None, 'Respawns', 'respawns', 'respawns', 'uwsgi.respawns', 'line'],
+        'lines': [
+            ['respawn_count', 'respawns', 'incremental']
+        ]
+    },
+}
+
+
+class Service(SocketService):
+    def __init__(self, configuration=None, name=None):
+        super(Service, self).__init__(configuration=configuration, name=name)
+        self.url = self.configuration.get('host', 'localhost')
+        self.port = self.configuration.get('port', 1717)
+        self.order = ORDER
+        self.definitions = deepcopy(CHARTS)
+
+        self.last_result = {}
+        self.workers = []
+
+    def read_data(self):
+        """
+        Read data from socket and parse as JSON.
+        :return: (dict) stats
+        """
+        raw_data = self._get_raw_data()
+        if not raw_data:
+            return None
+        try:
+            return json.loads(raw_data)
+        except ValueError as err:
+            self.error(err)
+            return None
+
+    def check(self):
+        """
+        Parse configuration, check if redis is available, and dynamically create chart lines data
+        :return: boolean
+        """
+        self._parse_config()
+
+        # Clear dynamic dimensions, these are added during `_get_data()` to allow adding workers at run-time
+        for chart in ('requests', 'tx', 'avg_rt', 'memory_rss', 'memory_vsz'):
```

Comment:
make a global var for `('requests', 'tx', 'avg_rt', 'memory_rss', 'memory_vsz')`